discord_bots/views/configure_category.py

```diff
@@ -0,0 +1,319 @@
+        await interaction.response.send_modal(modal)
+        return True
+
+    @button(label="Save", style=ButtonStyle.success, row=4)
+    async def save(self, interaction: Interaction, button: Button):
+        await interaction.response.defer()
+        await self.disable_children(interaction)
+
+        confirmation_buttons = ConfirmationView(interaction.user.id)
+        confirmation_buttons.message = await interaction.followup.send(
+            embed=Embed(
+                description=f"⚠️ Are you sure you want to save configurations for category **{self.category.name}**?⚠️",
+                colour=Colour.yellow(),
+            ),
+            view=confirmation_buttons,
+            ephemeral=True,
+        )
+        await confirmation_buttons.wait()
+        if not confirmation_buttons.value:
+            await self.enable_children(interaction)
+            return False
+        else:
+            self.value = True
+            self.stop()
+            return True
+
+    @button(label="Cancel", style=ButtonStyle.danger, row=4)
+    async def cancel(self, interaction: Interaction, button: Button):
+        self.stop()
+        return False
+
+
+class CategoryDecayModal(Modal):
+    def __init__(
+        self,
+        view: CategoryConfigureView,
+    ):
+        super().__init__(title="Decay Settings", timeout=30)
+        self.view: CategoryConfigureView = view
+        self.sigma_decay_amount: TextInput = TextInput(
+            label="Sigma decay amount",
+            style=TextStyle.short,
+            required=False,
+            placeholder=str(self.view.category.sigma_decay_amount),
+        )
+        self.sigma_decay_grace_days: TextInput = TextInput(
+            label="Sigma decay grace days",
+            style=TextStyle.short,
+            required=False,
+            placeholder=str(self.view.category.sigma_decay_grace_days),
+        )
+        self.sigma_decay_max_decay_proportion: TextInput = TextInput(
+            label="Sigma decay max decay proportion",
+            style=TextStyle.short,
+            required=False,
+            placeholder=str(self.view.category.sigma_decay_max_decay_proportion),
+        )
+        self.add_item(self.sigma_decay_amount)
+        self.add_item(self.sigma_decay_grace_days)
+        self.add_item(self.sigma_decay_max_decay_proportion)
+
+    async def on_submit(self, interaction: Interaction[Client]) -> None:
+        self.decay_amount: float | None = None
+        self.decay_grace_days: int | None = None
+        self.decay_max_decay_proportion: float | None = None
+
+        try:
+            if not str(self.sigma_decay_amount) == "":
+                self.decay_amount = float(str(self.sigma_decay_amount))
+            if not str(self.sigma_decay_max_decay_proportion) == "":
+                self.decay_max_decay_proportion = float(
+                    str(self.sigma_decay_max_decay_proportion)
+                )
+        except Exception:
```

Comment:
Might be a good idea to extract these validations into a BaseModal class or utils -- similar to the ones in `config.py`, since they're pretty generic and it would reduce a bit of duplication. Validation functions for int, float, and bool.